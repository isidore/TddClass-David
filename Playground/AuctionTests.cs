using System;
using ApprovalTests;
using ApprovalTests.Reporters;
using Microsoft.VisualStudio.TestTools.UnitTesting;

namespace Playground
{
    [TestClass]
    [UseReporter(typeof(DiffReporter))]
    public class AuctionTests
    {
        [TestMethod]
        public void TestCreateAuction()
        {
            // create a an action class
            // 
            var seller = new User("John","Knoll","jpknoll@notucdavis.com","great","nowhere");
            UserTest userTest = new UserTest();
            var (users, scott) = userTest.SetupScott(true);
            users.MakeSeller(scott);
            var startTime = DateTime.Now.AddSeconds(1.0);
            var endTime = DateTime.Now.AddSeconds(3.0);
            var auction = new Auction(scott, "item description", 0.10, startTime, endTime);

            Assert.AreEqual(auction.Seller, scott);
            Assert.AreEqual(auction.ItemDescription, "item description");
            Assert.AreEqual(auction.ItemPrice, 0.10, 0.01);
            Assert.AreEqual(auction.StartDateTime, startTime);
            Assert.AreEqual(auction.EndDateTime, endTime     );
            //

        }

    }
}         