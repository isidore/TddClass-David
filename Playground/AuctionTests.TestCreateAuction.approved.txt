Seller: FirstName: Scott
 LastName: Kirkland
 Email: srkirkland@ucdavis.edu
 UserName: srkirkland
 Password: givingService
 ItemDescription: item description
 ItemPrice: 0.1
 StartDateTime: 6/24/2020 6:11:35 PM
 EndDateTime: 6/24/2020 6:11:37 PM